src/Nethermind/Nethermind.JsonRpc/Modules/Parity/PeerNetworkInfo.cs

```diff
@@ -0,0 +1,38 @@
+//  Copyright (c) 2021 Demerzel Solutions Limited
+//  This file is part of the Nethermind library.
+// 
+//  The Nethermind library is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+// 
+//  The Nethermind library is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//  GNU Lesser General Public License for more details.
+// 
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+// 
+
+using Nethermind.Network;
+using Nethermind.Network.P2P;
+
+namespace Nethermind.JsonRpc.Modules.Parity
+{
+    public class PeerNetworkInfo
+    {
+        public string LocalAddress { get; set; }
+        public string RemoteAddress { get; set; }
+
+        public PeerNetworkInfo(Peer peer)
+        {
+            LocalAddress = peer.Node.Host;
+            RemoteAddress = peer.InSession != null ? peer.InSession.State != SessionState.New
+                    ? peer.InSession.RemoteHost : "Handshake"
+                : peer.OutSession != null ? peer.OutSession.State != SessionState.New
+                    ? peer.OutSession.RemoteHost : "Handshake"
+                : null;
```

Comment:
This is hard to read. Please add method per session as you duplicate logic

Reply:
Actually State > New condition changes nothing, as
0 - New
1 - HandshakeComplete
2 - Initialized
3 - DisconnectingProtocols
4 - Disconnecting
5 - Disconnected
Yesterday we wondered about it and established, that "Handshake" should be displayed for New, and RemoteHost for not New.
When written current way it's not possible to display "Handshake" to state other than New, if the condition is fulfilled (State != New), it displays RemoteHost, no matter if state is 1, 2, 3, 4 or 5, same would be with condition State > New.
I believe it should works like that, but correct me if for disconnecting/disconnected (3, 4, 5) it should display something different than RemoteHost. And for sure I will rewrite it more read-friendly way.